app/workers/facilities/facility_location_download.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Facilities
+  class FacilityLocationDownload
+    include Sidekiq::Worker
+
+    def perform(type)
+      @type = type
+      process_changes
+      process_deletes
```

Comment:
Can we possibly wrap this in an active_record transaction rollback in the event of a failure on any of the two? So if delete fails, but changes were successful we would rollback everything.